diesel/src/connection/mod.rs

```diff
@@ -45,6 +45,10 @@ pub trait Connection: SimpleConnection + Sized + Send {
     /// If there is already an open transaction,
     /// savepoints will be used instead.
     ///
+    /// If the transaction fails to commit due to a `SerializationFailure` or a
+    /// `ReadOnlyTransaction` rollback will be attempted. If it succeeds,
+    /// original error will be returned. If it fails, rollback error will be returned.
+    ///
```

Comment:
Just some small wording improvements. I think they apply in the other comments too

```suggestion
    /// If the transaction fails to commit due to a `SerializationFailure` or a
    /// `ReadOnlyTransaction` a rollback will be attempted. If the rollback succeeds,
    /// the original error will be returned, otherwise the error generated by the rollback
    /// will be returned. In the second case the connection should be considered broken 
    /// as it contains a uncommitted unabortable open transaction.
```